src/sql/workbench/services/notebook/browser/models/cell.ts

```diff
@@ -306,6 +309,24 @@ export class CellModel extends Disposable implements ICellModel {
 		}
 	}
 
+	public get showTextView(): boolean {
+		return this._showTextView;
+	}
+	public set showTextView(val: boolean) {
+		if (val !== this._showTextView) {
```

Comment:
The if isn't doing anything useful here - if the value is the same then setting it again isn't going to change anything

Reply:
Normally we do this because we fire the events and only want to do that when the value actually changes. Do you not need that here?

Reply:
Well then, I will need help to better understand this. I need to implement both showPreview and showTextView such that they can be toggled on and off. Do I simply need to change both lines 306 and 316 to set the value without the conditional?